### PR TITLE
dev: add admin repair table statement.

### PR DESCRIPTION
**Pull request by AilinKid**
### What is changed, added or deleted? 

add `admin repair table` statement for v4.0

### What is the related PR or file link(s)? 

- [x] Reference link(s): [#12046](https://github.com/pingcap/tidb/pull/12046)
- [ ] This PR is to align with:
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/sql/statements/admin.md [hunk 1]
```diff
@@ -104,6 +110,8 @@ admin show ddl jobs;
 
 - `ADMIN CHECK TABLE tbl_name [, tbl_name] ...`：用于对给定表中的所有数据和对应索引进行一致性校验，若通过校验，则返回空的查询结果；否则返回数据不一致的错误信息。
 
+- `ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：在极端情况下用于对存储层中的表的元信息进行非可信的覆盖（需要人为保证老的表元信息可以完全由 CREATE TABLE STATEMENT 提供)
```
Comment (zimulala) [inline dev/reference/sql/statements/admin.md]:
```suggestion
- `ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：在极端情况下用于对存储层中的表的元信息进行非可信的覆盖（需要人为保证老的表元信息可以完全由 `CREATE TABLE STATEMENT` 提供)。
```

**Comment by AilinKid** [comment]
> 说明一下要个配置文件配合使用？

好的

File: dev/reference/sql/statements/admin.md [hunk 2]
```diff
@@ -104,6 +110,8 @@ admin show ddl jobs;
 
 - `ADMIN CHECK TABLE tbl_name [, tbl_name] ...`：用于对给定表中的所有数据和对应索引进行一致性校验，若通过校验，则返回空的查询结果；否则返回数据不一致的错误信息。
 
+- `ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：在极端情况下用于对存储层中的表的元信息进行非可信的覆盖（需要人为保证老的表元信息可以完全由 `CREATE TABLE STATEMENT` 提供)。该语句需要打开配置文件项中的 `repair-mode` 开关，并且需要确保所修复的表名在 `repair-table-list` 名单中。
```
Comment (yikeke) [inline dev/reference/sql/statements/admin.md]:
```suggestion
- `ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：在极端情况下，用于对存储层中的表的元信息进行非可信的覆盖（需要人为保证原表的元信息可以完全由 `CREATE TABLE STATEMENT` 提供）。该语句需要打开配置文件项中的 `repair-mode` 开关，并且需要确保所修复的表名在 `repair-table-list` 名单中。
```
有俩问题：
1. 在咱们的文档中，配置文件描述里似乎并没有对 `repair-mode` 的说明？建议在“配置文件描述”这个文档里补充下内容。
2. `repair-table-list` 名单在咱们的文档中也没有搜到，这里是第一次提及，建议说明一下这个名单去哪里找或者怎么找。

Comment (AilinKid) [reply dev/reference/sql/statements/admin.md]: 好的，我看下配置文件部分

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 3]
```diff
@@ -71,6 +71,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：false
 + 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整型类型时，即使之后开启此开关也不支持对此列表删除主键。
 
+### `repair-mode`
+
++ 用于开启非可信修复模式，该模式下启动时，可以过滤 `repair-table-list` 名单中坏表的加载。
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 用于开启非可信修复模式，启动该模式后，可以过滤 `repair-table-list` 名单中坏表的加载。
```

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 4]
```diff
@@ -71,6 +71,18 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 默认值：false
++ 默认情况下，不支持修复语法，默认启动时会加载所有表信息。
+
+### `repair-table-list`
+
++ 配合 `repair-mode` 为 true 时使用，用于列出实例中需要修复的坏表的名单，该名单由 ["db.table",] 组成。
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
+ 配合 `repair-mode` 为 true 时使用，用于列出实例中需要修复的坏表的名单，该名单的写法为 ["db.table1","db.table2"...]。
```
建议这样示意名单写法，更加清楚

File: dev/reference/sql/statements/admin.md [hunk 5]
```diff
@@ -110,7 +110,7 @@ admin show ddl jobs;
 
 - `ADMIN CHECK TABLE tbl_name [, tbl_name] ...`：用于对给定表中的所有数据和对应索引进行一致性校验，若通过校验，则返回空的查询结果；否则返回数据不一致的错误信息。
 
-- `ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：在极端情况下用于对存储层中的表的元信息进行非可信的覆盖（需要人为保证老的表元信息可以完全由 `CREATE TABLE STATEMENT` 提供)。该语句需要打开配置文件项中的 `repair-mode` 开关，并且需要确保所修复的表名在 `repair-table-list` 名单中。
+- `ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：`ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：在极端情况下，用于对存储层中的表的元信息进行非可信的覆盖，“非可信”是指需要人为保证原表的元信息可以完全由 `CREATE TABLE STATEMENT` 提供。该语句需要打开配置文件项中的 `repair-mode` 开关，并且需要确保所修复的表名在 `repair-table-list` 名单中。
```
Comment (yikeke) [inline dev/reference/sql/statements/admin.md]:
```suggestion
- `ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：`ADMIN REPAIR TABLE tbl_name CREATE TABLE STATEMENT`：在极端情况下，用于对存储层中的表的元信息进行非可信的覆盖，“非可信”是指需要人为保证原表的元信息可以完全由 `CREATE TABLE STATEMENT` 提供。该语句需要打开配置文件项中的 [`repair-mode`](/dev/reference/configuration/tidb-server/configuration-file.md#repair-mode) 开关，并且需要确保所修复的表名在 [`repair-table-list`](/dev/reference/configuration/tidb-server/configuration-file.md#repair-table-list) 名单中。
```

Comment (AilinKid) [reply dev/reference/sql/statements/admin.md]: 嗯嗯，谢谢 keke，非常实用！

**Pull request merged**